test/parallel/test-http-agent-scheduling.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+function createServer(count) {
+  return http.createServer(common.mustCallAtLeast((req, res) => {
+    // Return the remote port number used for this connection.
+    res.end(req.socket.remotePort.toString(10));
+  }), count);
+}
+
+function makeRequest(url, agent, callback) {
+  http
+    .request(url, { agent }, (res) => {
+      let data = '';
+      res.setEncoding('ascii');
+      res.on('data', (c) => {
+        data += c;
+      });
+      res.on('end', () => {
+        process.nextTick(callback, data);
+      });
+    })
+    .end();
+}
+
+function bulkRequest(url, agent, done) {
+  const ports = [];
+  let count = agent.maxSockets;
+
+  for (let i = 0; i < agent.maxSockets; i++) {
+    makeRequest(url, agent, callback);
+  }
+
+  function callback(port) {
+    count -= 1;
+    ports.push(port);
+    if (count === 0) {
+      done(ports);
+    }
+  }
+}
+
+function defaultTest() {
+  const server = createServer(8);
+  server.listen(0, onListen);
+
+  function onListen() {
+    const url = `http://localhost:${server.address().port}`;
+    const agent = new http.Agent({
+      keepAlive: true,
+      maxSockets: 5
+    });
+
+    bulkRequest(url, agent, (ports) => {
```

Comment:
Might be a bit nicer to convert these into async functions with await syntax and to wrap this stack into a helper function so it can be reused in both tests.

Reply:
Thank you for the suggestion! Unfortunately, I don't have time for refactoring this test at the moment, is it ok if we leave it as is?

Reply:
@jasnell is not clear if you have an hard block on the tests being converted or not. Can we land this anyway?

Reply:
No hard block!